pkg/client/simple/vfsclientset/clientset.go

```diff
@@ -64,9 +64,9 @@ func (c *VFSClientset) ListClusters(ctx context.Context, options metav1.ListOpti
 }
 
 // ConfigBaseFor implements the ConfigBaseFor method of simple.Clientset for a VFS-backed state store
-func (c *VFSClientset) ConfigBaseFor(cluster *kops.Cluster) (vfs.Path, error) {
+func (c *VFSClientset) ConfigBaseFor(ctx context.Context, cluster *kops.Cluster) (vfs.Path, error) {
 	if cluster.Spec.ConfigBase != "" {
-		return vfs.Context.BuildVfsPath(cluster.Spec.ConfigBase)
+		return vfs.FromContext(ctx).BuildVfsPath(cluster.Spec.ConfigBase)
```

Comment:
I'm struggling to see why the `vfs.Path` should depend on a passed-in `context.Context`. And only when the `ConfigBase` is set.

Shouldn't this either take a VFSContext or hang the VFSContext off of the VFSClientset?

Reply:
It looks like a VFSContext is closer to a Clientset than it is to a context.Context.

Reply:
> It looks like a VFSContext is closer to a Clientset than it is to a context.Context.

Yes, I think that's fair.  For example, if we had two different identities (which we don't, but if we did)... then I would expect we would have implement that with different VFSContext instances.

We could rename VFSContext - perhaps VFSScope ?

Our current use-case is for tests, so I didn't want to do _too_ much refactoring, but it prompts introducing contexts so perhaps that ship has sailed.  I do think that introducing contexts is good, even if we're not using the tracing / logging etc that it enables yet.

Reply:
It's sort of a `VFSClient`. It's a pity we have `VFSClientSet`; perhaps that should be `VFSSimpleClientset`?

Reply:
I don't have an objection to introducing `context.Context`s; it's just that this doesn't seem an appropriate use of them.

Reply:
That's reasonable - there's certainly an element of expedience in my thinking here.  If we add context.Context to a bunch more functions, then I think we can do it the more logical way you're suggesting.  I was trying to show some value before we added context.Context everywhere, but it sounds like we're already bought in to / resigned to the need for context.Context!